framework/src/iotbus/iotbus_pwm.c

```diff
@@ -192,7 +191,7 @@ int iotbus_pwm_get_duty_cycle(iotbus_pwm_context_h pwm)
 		zdbg("ioctl(PWMIOC_GETCHARACTERISTICS) failed: %d\n", errno);
 		return IOTBUS_ERROR_UNKNOWN;
 	}
-	return (int)((info->duty * 100.0 / 65536) + 0.5);
+	return (int)((info->duty >> 16) * 100.0 + 0.5);
```

Comment:
This is wrong. The duty is a int type so that it will be zero before calculating "* 100 + 0.5".

Reply:
will keep original changes.